docs/Middleware-Guide.md

```diff
@@ -0,0 +1,82 @@
+3. After the latest time until which the staker is obliged to participate with the current stake passes, the staker now can complete its queued withdrawal by calling on the `completeQueuedWithdrawal(..)` function in `StrategyManager.sol`.
+
+The following figure illustrates the above flow: 
+![Staker deregistering](./images/staker_withdrawing.png)
+
+
+### *Slashing*
+As mentioned above, EigenLayer is built to support slashing as a result of an on-chain-checkable, objectively attributable action. In order for a middleware to be able to slash an operator in an objective manner, the middleware needs to deploy a DisputeResolution contract where anyone would be able to raise a challenge against an EigenLayer operator for its adversarial action. On successful challenge, the DisputeResolution contract calls the `freezeOperator(..)` function in ServiceManager contract which in turn accesses the `freezeOperator(..)` function in the EigenLayer's Slasher contract. On successful execution of this API, the event `OperatorFrozen(..)` is emitted.  
+
+The following figure illustrates the above flow: 
+![Slashing](./images/slashing.png)
+
+
+
+
+## Guide To Provided Middleware Contracts:
+The EigenLayer team has built a set of reusable and extensible contracts for use in middlewares built on top of EigenLayer. These are contained in the general-purpose /middleware/ folder, which contains code that can be extended, used directly, or consulted as a reference in building middleware on top of EigenLayer. There are several basic contracts that all middleware-specific contracts can be built on:
+- The *RegistryBase contract* is a basic registry contract that can be used to track operators opted into running a middleware.  Importantly, this base registry contract assumes a maximum of two quorums, where each quorum represents an aggregation of a certain type of stake.  Middlewares may want to weigh their rewards based on the type of stake delegated to their operators and weigh each quorum differently.  
+- The *VoteWeigherBase contract* tracks an operator’s “weight” in a given quorum, across all strategies that are associated with that quorum.  This contract also manages which contracts are categorized under which quorums - this includes both removing and adding functionalities as well as changing strategy weights.  
+
+It’s expected that many middlewares will require a quorum of registered operators to sign on commitments. A quorum is a group of stakers who opt into a service while satisfying a particular trait. Examples of a trait could be stETH stakers or native stakers.  To this end, we have developed a set of contracts designed to optimize the cost of checking signatures through the use of a BLS aggregate signature scheme:
```

Comment:
here's **an actual definition of quorum** -- seems overly-strict to me, but something like this should be up in the terminology/definitions section.

Reply:
loool burned. But I think the idea here was to say that as an example, we have a BLS signature based registry (aka builds around sig aggregation).  I think the "to this end" makes no sense, I have revised it.